infra/scoped/secrets.tf

```diff
@@ -171,3 +176,18 @@ resource "aws_secretsmanager_secret_version" "smoke_test-auth0_credentials" {
   secret_id     = aws_secretsmanager_secret.smoke_test-auth0_credentials.id
   secret_string = jsonencode(local.smoke_test_credentials)
 }
+
+# Buildkite
+
+resource "aws_secretsmanager_secret" "buildkite-auth0_credentials" {
+  name = "identity/${terraform.workspace}/buildkite/credentials"
+
+  tags = {
+    "Name" = "identity/${terraform.workspace}/buildkite/auth0_client_secret"
+  }
+}
+
+resource "aws_secretsmanager_secret_version" "buildkite-auth0_credentials" {
+  secret_id     = aws_secretsmanager_secret.buildkite-auth0_credentials.id
+  secret_string = jsonencode(local.buildkite_credentials)
+}
```

Comment:
FYI, we have a prebuilt secrets module now: https://github.com/wellcomecollection/terraform-aws-secrets